### PR TITLE
Introduce launcher 

### PR DESCRIPTION
**Pull request by zhangsoledad**
this PR mainly simplified the launch code
https://github.com/nervosnetwork/ckb/blob/853eaf5ab5f5a9b678f496abb74cf576a3501804/ckb-bin/src/subcommand/run.rs#L11-L62

### CODE REVIEW

File: shared/src/lib.rs [hunk 1]
```diff
@@ -1,10 +1,6 @@
 //! TODO(doc): @quake
 
 // num_cpus is used in proc_macro
```
Comment (yangby-cryptape) [inline shared/src/lib.rs:3]:
```suggestion
```

**Review by yangby-cryptape — APPROVED** [review]

**Review by quake — DISMISSED** [review]

**Review by yangby-cryptape — DISMISSED** [review]

**Review by quake — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors r=yangby-cryptape,quake

**Comment by zhangsoledad** [comment]
bors retry

**Comment by zhangsoledad** [comment]
bors retry

**Review by yangby-cryptape — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Comment by yangby-cryptape** [comment]
bors r=yangby-cryptape,quake

**Pull request merged**